graph/src/blockchain/block_stream.rs

```diff
@@ -80,8 +82,30 @@ pub trait BlockStream<C: Blockchain>:
 {
 }
 
-pub trait BlockStreamBuilder<C: Blockchain> {
-    fn build() -> Box<dyn BlockStream<C>>;
+/// BlockStreamBuilder is an abstraction that would separate the logic for building streams from the blockchain trait
+/// the goal is that this api is as close as possible to existing implementations in order to make the refactor easier.
```

Comment:
This second line of the comment is useful as context for this PR, but I wouldn't keep it as a permanent comment since then it's not clear what refactor this is talking about.